src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -1425,6 +1483,27 @@ private Status TryScanMethod(MethodIL methodIL, Value[] parameters, Stack<Method
                     case ILOpcode.ldind_u4:
                     case ILOpcode.ldind_i8:
                         {
+                            if (opcode == ILOpcode.ldobj)
+                            {
+                                TypeDesc type = methodIL.GetObject(reader.ReadILToken()) as TypeDesc;
+                                opcode = type.Category switch
+                                {
+                                    TypeFlags.SByte => ILOpcode.ldind_i1,
+                                    TypeFlags.Boolean or TypeFlags.Byte => ILOpcode.ldind_u1,
+                                    TypeFlags.Int16 => ILOpcode.ldind_i2,
+                                    TypeFlags.Char or TypeFlags.UInt16 => ILOpcode.ldind_u2,
+                                    TypeFlags.Int32 => ILOpcode.ldind_i4,
+                                    TypeFlags.UInt32 => ILOpcode.ldind_u4,
+                                    TypeFlags.Int64 or TypeFlags.UInt64 => ILOpcode.ldind_i8,
```

Comment:
Add Float and Double? They are handled in the switch statement below.

Reply:
Single/double added.